eclair-core/src/main/scala/fr/acinq/eclair/message/Postman.scala

```diff
@@ -102,13 +102,40 @@ object Postman {
             case Failure(f) =>
               replyTo ! MessageFailed(f.getMessage)
             case Success((nextNodeId, message)) =>
-              if (replyPath.isEmpty) { // not expecting reply
-                sendStatusTo += (messageId -> replyTo)
-              } else { // expecting reply
-                subscribed += (messageId -> replyTo)
-                context.scheduleOnce(timeout, context.self, Unsubscribe(messageId))
+              if (nextNodeId == nodeParams.nodeId) {
+                OnionMessages.process(nodeParams.privateKey, message) match {
```

Comment:
Shouldn't this be done in `OnionMessages.buildMessage` instead? It looks like it would be more natural there, we would start by checking if we are the introduction node of the `destination`'s blinded path:

- if yes, we decrypt the first hop of that route
- otherwise, we create a new blinded path towards the `destination` as currently done

Also, I'm not sure we correctly handle the case where we're asked to send a message to ourselves? Is this something we actually need to handle explicitly, or will it never happen?

Reply:
We don't handle the sending to ourselves scenario indeed. I just don't think there is a reason to do that.

Reply:
No need to explicitly handle it then if it's not an issue. What do you think of doing the blinded path unwrapping in `buildMessage`?

Reply:
Yes, it could be done in `OnionMessages.buildMessage` too, I just thought it was easier to it like that. I can refactor it inside `OnionMessages.buildMessage` if you prefer.

Reply:
I think it would make more sense there, because any caller of `buildMessage` wants the first blinded hop to be unwrapped if we're the introduction node, right?

Reply:
Done